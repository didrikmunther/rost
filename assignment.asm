	; [header]
	global main
	extern printf

	section .text
main:
	mov rax, 5
	push rax ; let a = 5

	mov rax, [rsp] 	;	 a
	add rax, 2		; + 2

	push rax 		; let b = ...

	mov rax, [rsp+8] 	; b
	add rax, [rsp]		; + a
	add rax, 1			; + 1
	push rax			; let c = ...

	; [procedure 0]
	mov rdi, _data_0
	mov rsi, [rsp] 		; c
	xor rax, rax
	call printf
	; [exit]

	pop rbx
	pop rbx
	pop rbx

	ret	; [exit program]
	; [data]

	section .data
_data_0:
	db "%i", 0