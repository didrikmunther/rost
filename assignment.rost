let a = 100
let b = 10
let c: int = a + b
let d = 5

printf("%i %i", c + 1 + d + c, c, c, c)