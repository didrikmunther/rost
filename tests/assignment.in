let a = 123
let b = 321
printf("%i\n", a)
a = 1
printf("%i\n", a + b)
printf("%i\n", b)