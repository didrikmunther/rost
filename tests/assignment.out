123
322
321